File: NeetCode/Arrays & Hashing/Contains Duplicate.js

/**
 * Time Complexity O(nlogn)
 * Using Sorting approach
 */ 

class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        nums.sort();
        for(let i = 0; i < nums.length; ++i)
        {   
            if (nums[i] == nums[i+1])
                return true; 
        }
        return false;
        }
}

File: NeetCode/Arrays & Hashing/Contains Duplicate.cpp
#include <iostream>
#include <vector>

/**
 * Time Complexity O(n ^ 2)
 * Using Brute Force approach
 */
class Solution
{
public:
    bool hasDuplicate(std::vector<int> &inputArray)
    {
        for (int i = 0; i < inputArray.size(); i++)
        {
            for (int j = i + 1; j < inputArray.size(); j++)
            {
                if (inputArray[i] == inputArray[j])
                    return true;    
            }
        }
        return false;
    }
};
